modules/porous_flow/src/materials/PorousFlowMaterialPermeabilityFromPorosity.C

```diff
@@ -0,0 +1,82 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "PorousFlowMaterialPermeabilityFromPorosity.h"
+
+template<>
+InputParameters validParams<PorousFlowMaterialPermeabilityFromPorosity>()
+{
+  InputParameters params = validParams<Material>();
+
+  MooseEnum poroperm_function("kozeny_carman_fd2=0 kozeny_carman_phi0=1", "kozeny_carman_fd2");
+  params.addParam<MooseEnum>("poroperm_function", poroperm_function, "Function relating porosity and permeability. The options are: kozeny_carman_fd2 = f d^2 phi^n/(1-phi)^m (where phi is porosity, f is a scalar constant with typical values 0.01-0.001, and d is grain size). kozeny_carman_phi0 = k0 (1-phi0)^m/phi0^n * phi^n/(1-phi)^m (where phi is porosity, and k0 is the permeability at porosity phi0)");
+  params.addParam<Real>("k0", "The permeability scalar value (usually in m^2) at the reference porosity, required for kozeny_carman_phi0");
+  params.addParam<RealTensorValue>("k_anisotropy", "A tensor to multiply the calculated scalar permeability, in order to obtain anisotropy if required. Defaults to isotropic permeability if not specified.");
+  params.addParam<Real>("phi0", "The reference porosity, required for kozeny_carman_phi0");
+  params.addParam<Real>("f", "The multiplying factor, required for kozeny_carman_fd2");
+  params.addParam<Real>("d", "The grain diameter, required for kozeny_carman_fd2");
+  params.addRequiredParam<Real>("n", "Porosity exponent (numerator)");
+  params.addRequiredParam<Real>("m", "(1-porosity) exponent (denominator)");
+  params.addRequiredParam<UserObjectName>("PorousFlowDictator_UO", "The UserObject that holds the list of Porous-Flow variable names.");
+  params.addClassDescription("This Material calculates the permeability tensor from a specified function of porosity");
+  return params;
+}
+
+PorousFlowMaterialPermeabilityFromPorosity::PorousFlowMaterialPermeabilityFromPorosity(const InputParameters & parameters) :
+    DerivativeMaterialInterface<Material>(parameters),
+    _k0_set( parameters.isParamValid("k0") ),
```

Comment:
Spacing here is different than below.  I suggest using _k0_set(parameters.isParamValid("k0"))  with no spaces - seems to be MOOSE convention.  Sorry for being so pedantic, but when you put this all into MOOSE you'll get the same comment!